lib/bundler/definition.rb

```diff
@@ -709,12 +709,8 @@ def converge_locked_specs
       # and Gemfile.lock. If the Gemfile modified a dependency, but
       # the gem in the Gemfile.lock still satisfies it, this is fine
       # too.
-      locked_deps_hash = @locked_deps.inject({}) do |hsh, dep|
-        hsh[dep] = dep
-        hsh
-      end
       @dependencies.each do |dep|
-        locked_dep = locked_deps_hash[dep]
```

Comment:
@segiddins I ended up opting for the next line `locked_dep = nil unless locked_dep == dep` to compensate for the inequality. This allows us to use the `O(1)` lookup of the `@locked_deps` hash without creating an entirely new hash.